### PR TITLE
[DO NOT MERGE FOR NOW] Add HeadCategory to back-end (issues #271, #285, & #291)

### PR DESCRIPTION
**Pull request by acharliekelly**
Adds ability to read 'Head Category' (Column D, fka Category Groups) from Google sheet, as well as changing every instance of "Category Group" to "Head Category". No changes to front end.

Note that column name will need to be updated at the same time that this PR is committed, otherwise it will cause errors trying to parse the sheet.

### CODE REVIEW

**Comment by RajChaudhry** [comment]
On my machine it's infinitely loading all the resources. Does it work on yours?

File: src/reducers/resourceReducer.js [hunk 1]
```diff
@@ -22,6 +22,27 @@ export function isFetchingResource(state = initialState.isFetchingResource, acti
     }
 }
 
+export function headCategories(state = initialState.headCategories, action) {
+  switch (action.type) {
```
Comment (galiat) [inline src/reducers/resourceReducer.js]: Would be good to avoid duplication & let `headCategories()` & `categories()` to share code by extracting a method

Comment (galiat) [reply src/reducers/resourceReducer.js]: could be another PR

File: src/reducers/resourceReducer.js [hunk 2]
```diff
@@ -22,6 +22,27 @@ export function isFetchingResource(state = initialState.isFetchingResource, acti
...
+    case types.LOAD_RESOURCE_DATA_SUCCESS:
+      const headCategories = {};
+      for (let data of action.resource) {
+        let headCat = data.headcategories.split(',');
+        headCat.forEach(group => headCategories[group] = group.trim());
```
Comment (galiat) [inline src/reducers/resourceReducer.js]: transcribing IRL convo: change this from group to something else; will be kinda a non-issue with refactor

**Comment by ndesai06** [comment]
Update 6/11: Remove "(TESTING) Category Groups" entirely from code and database. Change "Category (Autosort Script)" to "Head Category" and reflect in code and database.

**Comment by RajChaudhry** [comment]
Done via #324 .

**Pull request closed without merging**